01_clean_dataset.R

```diff
@@ -0,0 +1,134 @@
+library(tidyverse)
+library(janitor)
+library(readxl)
+library(googlesheets4)
+library(glue)
+library(googleLanguageR)
+
+ks <- read_sheet(ss = Sys.getenv("SYR_EQUAKE_GS_URL"),sheet = "survey")
+
+dat_fp <- file.path(
+    Sys.getenv("SYR_EQUAKE_DIR"),
+    "kobo_data_incoming",
+    "Syria_earthquake_-_shelters_multi-sectoral_assessment_-_latest_version_-_False_-_2023-02-17-07-19-17.xlsx"
+) 
+
+dat <- read_excel(dat_fp)
+
+##########################
+#### CLEANING COLUMNS ####
+##########################
+
+# get names for conversion
+
+ks_names <- ks %>%
+    filter(
+        !is.na(name_new),
+        !str_detect(name_new, "eng$")
+    )
+
+# create named vector for use in str_replace_all
+reps <- ks_names$name_new
+names(reps) <- ks_names$name
+
+# simply pivot all data longer and use str_replace_all to rename vectors
+# then re-pivot everything to wide data and re-convert
+df_clean <- dat %>%
+    mutate(
+        id = row_number(),
+        across(
+            .cols = everything(),
+            .fns = as.character
+        )
+    ) %>%
+    pivot_longer(
+        cols = -id
+    ) %>%
+    mutate(
+        name = str_replace_all(
+            name,
+            rev(reps) # reverse so substrings don't create problems
+        )
+    ) %>%
+    pivot_wider() %>%
+    type_convert() %>%
+    select(
+        matches(paste(reps, collapse = "|")),
+        -ends_with("sm")
+    )
```

Comment:
This throws the following warning  and creates list-cols. I don't think this is intended?


``` 
Warning message:
Values from `value` are not uniquely identified; output will contain list-cols.
* Use `values_fn = list` to suppress this warning.
* Use `values_fn = {summary_fun}` to summarise duplicates.
* Use the following dplyr code to identify duplicates.
  {data} %>%
    dplyr::group_by(id, name) %>%
    dplyr::summarise(n = dplyr::n(), .groups = "drop") %>%
    dplyr::filter(n > 1L) 

````

Reply:
Good catch! Happened because I added some rows to highlight our calculated columns in the Google Sheet, so it was inputting an NA value into our replacements in `str_replace_all()` and converting all names to `NA`. This has fixed it.